man/internal.selfref.rd

```diff
@@ -0,0 +1,37 @@
+\name{.internal.selfref}
+\alias{.internal.selfref}
+\title{Internal Self-Reference Attribute in data.table}
+\description{
+  The \code{.internal.selfref} attribute is an internal mechanism used by \code{data.table} to optimize memory management and performance. It acts as a pointer that allows \code{data.table} objects to reference their own memory location. While the \code{.internal.selfref} attribute may appear to always point to \code{NULL} when inspected directly, this is a result of its implementation in R's memory management system. The true significance of this attribute lies in its role in supporting reference semantics, which enables efficient in-place modification of \code{data.table} objects without unnecessary copying.
```

Comment:
It's not that the external pointer _appears_ to point at `NULL`; that's [done deliberately](https://github.com/Rdatatable/data.table/blob/9875906f445eddfb2a67b6326f6ce66537dcd374/src/assign.c#L33) so that `identical()` would be `TRUE` for such attributes from two different `data.table`s. (Why is that needed? So that `identical()` would work on `data.table`s with otherwise identical contents.) The actual self-reference pointer lives in the `prot` part of the external pointer, visible using `.Internal(inspect(x))`, wrapped into another external pointer to avoid creating a reference loop visible to R. Once the `data.table` is duplicated, its address will be different from what is stored in the self-reference attribute, making it possible to detect such copies.

Try summarising the [comments and the code of `_selfrefok`](https://github.com/Rdatatable/data.table/blob/9875906f445eddfb2a67b6326f6ce66537dcd374/src/assign.c#L45-L139) to document what it actually does.